test/k8sT/hubble.go

```diff
@@ -32,149 +32,147 @@ import (
 )
 
 var _ = Describe("K8sHubbleTest", func() {
+	// We want to run Hubble tests both with and without our kube-proxy
+	// replacement, as the trace events depend on it. We thus run the tests
+	// on GKE and our 4.9 pipeline.
```

Comment:
Comment says 4.9 and GKE but we seem to check for not running on NetNext or or 4.19 (`RunsOnNetNextOr419Kernel`) which, I guess, is the complement set. Maybe that's how we do this thing, but I guess it can be somewhat confusing and might have unexpected results if we add kernels.

Reply:
I was going to make that change, but after some thoughts, I'm not sure it makes sense:
We don't have a `RunsOn49Kernel()` helper because we don't pass a flag for 4.9. 4.9 in our CI is literally defined as the lack of either `KERNEL=419` or `NETNEXT=1`. So RunsOn49Kernel() would really be `DoesNotRunOnNetNextOr419Kernel()`. With the GKE special case, that would bring us to write:
```go
SkipContextIf(func() bool {
	return !helpers.RunsOnGKE() && helpers.RunsOnNetNextOr419Kernel()
}, ...
```
which is exactly equivalent to:
```go
SkipContextIf(helpers.RunsOnNetNextOr419Kernel, ...
```

When we add new kernels (e.g., `KERNEL=54`) I think we'll have to rework that set of helpers anyway.

Reply:
I don't understand that infrastructure that well, and I think leaving it as it is is perfectly fine since the comment above explains it.

Tangent: as a user, I would love to avoid using env variable at the ginkgo level and just pass options. I haven't had a chance to look how easy the conversion would be, but if we ever decide to add another kernel switching to options (e.g., cilium.kernel or cilium.environment) instead of env vars might be something to consider.

Reply:
> Tangent: as a user, I would love to avoid using env variables (especially, differently named ones) at the ginkgo level and just pass options. I haven't had a chance to look how easy the conversion would be

I agree. It should be fairly trivial to implement by following [one of our existing options](https://github.com/cilium/cilium/blob/e8605f88baabceebef52170b7337c793436492aa/test/config/config.go#L33). Just needs someone to invest cycles.